controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -907,7 +919,7 @@ private long findSegmentSplitsMerges(List<StreamSegmentRecord> referenceSegments
                                     activeEpochRecord.getSegments().stream()
                                                           .map(x -> newSegmentRecord(computeSegmentId(getSegmentNumber(x.segmentId()),
                                                                   committingTxnRecord.getNewActiveEpoch()),
-                                                                  timeStamp, x.getKeyStart(), x.getKeyEnd()))
+                                                                  timeStamp + 1, x.getKeyStart(), x.getKeyEnd()))
```

Comment:
Why +1?

Reply:
in rolling txn, we create two epochs -> txnDuplicateEpoch, activeDuplicateEpoch..
We want to have our time monotonically increasing. 
Currently they both get the same `time`.

Note: time is only used for UI queries for scaleMetadata which results in calling getEpochsAtTime.
So it doesnt affect any other business logic.. But while testing, having monotonically increasing time gives us an exact result for each value of time.